client/src/main/java/com/vesoft/nebula/encoder/NebulaCodecImpl.java

```diff
@@ -213,21 +213,18 @@ public SchemaProviderImpl genSchemaProvider(long ver, Schema schema) {
         if (names == null || values == null || names.size() != values.size()) {
             throw new RuntimeException("NebulaCodeImpl input wrong value");
         }
-
-        TagItem tag = metaCache.getTag(spaceName, schemaName);
         Schema schema;
         long ver;
-        if (tag == null) {
+        try {
+            TagItem tag = metaCache.getTag(spaceName, schemaName);
+            schema = tag.getSchema();
+            ver = tag.getVersion();
+        } catch (IllegalArgumentException e) {
             EdgeItem edge = metaCache.getEdge(spaceName, schemaName);
```

Comment:
here also can throw exception

Reply:
Then exception will be thrown.
If IllegalArguementException is thrown, schemaName is not exist.

Reply:
better unify handle the exception, the exception here should no return the edge exception.here return the error msg is no only edge.

Reply:
OK. Catch getEdge operation for IllegalArgumentException, and thrown new IllegalArgumentException with "schemaName does not exist" msg.